### PR TITLE
Chanllenge

### PR DESCRIPTION
**Pull request by bryanhmiller**
Please note that the branch is spelled "chanllenge", as opposed to the more traditional "challenge."
@morecallan 

Description of what happens when you click the button:

After the user has entered a values for the html inputs, and clicks the html "Grow Tree", JavaScript fires through the use of an eventListener when the button is clicked.  The eventListener receives the input through a variable set to getElementById, the variable is appended with "addEventListener" specifying the type of event and action to be taken.  The function passTreeSpecs is called, in turn calling the functions myTreeChar, myTreeHeight and growTree (with the single object treeSpecs passed into this function.)  The function myTreeChar uses getElementById to replace the value of key "char" (originally set to an empty string in the object treeSpecs) with the user's input.  Then the function myTreeHeight is called, which likewise uses getElementById to replace the value of key "height" (orginally set to 0 in the object treeSpecs) with the user's input.  Finally, the function growTree is passed the object treeSpecs freshly updated with the user's specific input.  The growTree function then uses the less specific userTreeSpecs to retrieve the values of height (userTreeSpecs.height) and char (userTreeSpecs.char) from the single object treeSpecs to supply the necessary values to run the nested for loops in the function growTree.  The console.log at on the outer loop then iteratively logs the information obtained to the console, thus producing a tree.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		for (var k = 0; k < userTreeSpecs.height - (i + 1); k++) {
+			air += " ";
+		}
+		for (var q = 0; q < ( 2 * i ) + 1; q++ ) {
+			charTree += userTreeSpecs.char;
+		}
+		console.log(air + charTree);
+	}
+}
+// If either of the input fields does not have a value in it when the user
```
Comment (morecallan) [inline main.js]: Make sure you are following this instruction.

**Review by morecallan — COMMENTED** [review]
@bryanhmiller Really really close! I like that you went back and typed up the notes. I can tell it sunk in a bit, too. Your code is well commented and your commit history is strong, just make sure to see my one note about following all project instructions.

**Comment by bryanhmiller** [comment]
@morecallan I have added the missing alert component.  I believe it now meets all the requirements.  Thanks so much.

**Comment by morecallan** [comment]
👍 let's talk saturday about the alert mystery!

**Pull request merged**